### PR TITLE
Limit the plan generation of Hypervisor to the machines passed.

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- currently the hypervisor is generating the plan for all hypervisor machines not considering the cli filter that can be specific to some machines or azs
- changed the eq method of HypervisorGroup to also compare the app_units

### CODE REVIEW

File: cou/steps/hypervisor.py
Comment (rgildein) [inline cou/steps/hypervisor.py:161]: not-blocking: we are not using machine anywhere else, so I think we should use `if unit.machine in self.machines:`.

File: cou/steps/hypervisor.py
Comment (rgildein) [inline cou/steps/hypervisor.py:162]: The note above should be moved to this part, since it belong to it.

File: cou/steps/hypervisor.py
Comment (rgildein) [inline cou/steps/hypervisor.py:161]: I think we should log skipped machine, so I'm maybe doing something like this:

```python
for unit in app.units.values():
    if unit.machine not in self.machines:
        logger.debug("skipping machine %s", unit.machine)
        continue

    az = ...
```
What do you say?

File: cou/steps/plan.py
Comment (rgildein) [inline cou/steps/plan.py:364]: I think we do not need this anymore, since we are already going through `analysis_result.apps_data_plane`, or not? We only need to drop `ceph-osd` form those apps?

File: tests/unit/steps/test_hypervisor.py
Comment (rgildein) [inline tests/unit/steps/test_hypervisor.py:57]:
```suggestion
    machines = [COUMachine(f"{i}", (), f"az{i}") for i in range(3)]
```

File: tests/unit/steps/test_hypervisor.py
Comment (rgildein) [inline tests/unit/steps/test_hypervisor.py:40]:
```suggestion
    machines = [COUMachine(f"{i}", (), f"az{i}") for i in range(3)]
```

File: tests/unit/steps/test_hypervisor.py
Comment (rgildein) [inline tests/unit/steps/test_hypervisor.py:74]: Can you create group3, which will be same as group1? And compare them.

File: tests/unit/steps/test_hypervisor.py
Comment (rgildein) [inline tests/unit/steps/test_hypervisor.py:295]: not-blocking: `test_hypervisor_upgrade_plan_filtered` -> `test_hypervisor_upgrade_plan_single_machine` or maybe even drop `test_hypervisor_plan_single_unit` (Meaning we need to do same for test above).

**Review by rgildein — CHANGES_REQUESTED** [review]
Overall it's looks good, only some small changes required.

File: cou/steps/plan.py
Comment (gabrielcocenza) [reply cou/steps/plan.py:364]: Maybe in the future we'll also need to drop subordinates, but I guess it's ok drooping now :+1:

File: cou/steps/hypervisor.py
Comment (rgildein) [inline cou/steps/hypervisor.py:164]: not-blocking: lines 158-159 should be moved before this line

File: cou/steps/plan.py
Comment (rgildein) [reply cou/steps/plan.py]: yes, we will need to do that, but now it's fine

**Review by rgildein — APPROVED** [review]

**Review by rgildein — APPROVED** [review]

File: cou/steps/plan.py
Comment (agileshaw) [inline cou/steps/plan.py:361]: This can simply be:
```
hypervisor_apps = [app for app in analysis_result.apps_data_plane if app.charm != "ceph-osd"]
```

**Review by agileshaw — CHANGES_REQUESTED** [review]
Overall looks good, just one suggestion.

File: cou/steps/plan.py
Comment (rgildein) [reply cou/steps/plan.py:361]: There is a huge chance that we will need to filter more than just ceph-osd, so I do not think that this should be blocked by this. Resolving this conversation.

Comment (agileshaw) [reply cou/steps/plan.py:361]: The suggested approach can handle additional filtering as well. But I'm ok to unblock this PR with the current approach.

**Review by agileshaw — APPROVED** [review]

**Review by rgildein — APPROVED** [review]

**Pull request merged**
